### PR TITLE
Update documentation to not refer to experimental Gateway APIs

### PR DESCRIPTION
**Pull request by craigbox**
https://github.com/istio/istio.io/issues/15043#issuecomment-2105059720

> Oh but we put that everywhere. We should really not do that. A page like https://preliminary.istio.io/latest/docs/tasks/traffic-management/request-routing/ has no business telling users to turn on alpha!! it doesn't need it.

> What we should do is on the few pages that do, say "This feature of GW API is alpha. We recommend you don't use it yet. If you do want to, here is how..."

I went through all the pages that referred to the `gateway-api-gamma-support` boilerplate, and did one of two things:

- update to refer to `gateway-api-support`, which installs the 1.10 CRDs that include the GAMMA APIs (if using `Gateway`/`HTTPRoute`)
- updated to refer to a renamed `gateway-api-gamma-experimental` (if using `TLSRoute` or `TCPRoute`)

### CODE REVIEW

**Comment by ericvn** [comment]
Also, @frankbu has done a lot of the documentation around Gateway APIs.

**Review by howardjohn — APPROVED** [review]

File: content/en/boilerplates/gateway-api-gamma-experimental.md [hunk 1]
```diff
@@ -6,10 +6,8 @@
 {{< /tip >}}
 
 {{< warning >}}
-This document configures internal mesh (east-west) traffic
-that requires Gateway API features that are either
+This document configures Istio using Gateway API features that are
```
[the remark this answers is not in the document]
Comment (craigbox) [reply content/en/boilerplates/gateway-api-gamma-experimental.md:9]: This boilerplate also #include's gamma-api-choose.md which says

> The following instructions allow you to choose to use either the Gateway API or the Istio configuration API when configuring traffic management in the mesh. Follow instructions under either the `Gateway API` or `Istio APIs` tab,
according to your preference.

I'm happy with that being explicit enough (at least for now)

**Comment by craigbox** [comment]
@frankbu The demo/minimal profiles pass but 'none' fails.  Do you have any thoughts on why?

**Comment by ericvn** [comment]
Looking at the failure in `/tasks/traffic-management/ingress/ingress-sni-passthrough/gtwapi_test.sh ` I see a 
```
bash: line 32: bpsnip_gateway_api_gamma_support_enable_alpha_crds: command not found
```
and later a 
```
VERIFY FAILED snip_deploy_an_nginx_server_5 (timeout after 120s):
received:
"Error from server (BadRequest): container istio-proxy is not valid for pod my-nginx-79c8cf654c-h4rhv"
expected:
"subject: CN=nginx.example.com"
```

**Comment by ericvn** [comment]
Maybe just need an update in the script to the new snippet name c/support/experimental?

**Comment by craigbox** [comment]
Sigh, I didn't see that because I was looking at `tcp-traffic-shifting`. 🤦 

Because I don't do this often I don't remember how to get from https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15150/doc.test.profile-none_istio.io/1792503745053986816 to the actual script log where you would have seen this error.

This line is in the output:

`2024-05-20T11:36:07.947409Z	info	tf	Wrote trace to https://gcsweb.istio.io/gcs/istio-prow/pr-logs/pull/istio_istio.io/15150/doc.test.profile-none_istio.io/1792503745053986816/artifacts/trace.yaml`

but I'd really like if a link to whatever file you found that error in was there instead :)

**Review by ericvn — APPROVED** [review]

**Pull request merged**